src/backend/tests/mocked/slackMessages.test.ts

```diff
@@ -0,0 +1,464 @@
+import { Organization, User } from '@prisma/client';
+import { createSlackMessageEvent, createTestOrganization, createTestUser, resetUsers } from '../test-utils';
+import {
+  batmanAppAdmin,
+  batmanSettings,
+  supermanAdmin,
+  supermanSettings,
+  wonderwomanGuest,
+  wonderwomanSettings
+} from '../test-data/users.test-data';
+import * as apiFunctions from '../../src/integrations/slack';
+import AnnouncementService from '../../src/services/announcement.service';
+import slackServices from '../../src/services/slack.services';
+import { vi } from 'vitest';
+import { HttpException } from '../../src/utils/errors.utils';
+
+vi.mock('../../src/integrations/slack', async (importOriginal) => {
+  return {
+    ...(await importOriginal<typeof import('../../src/integrations/slack')>()),
+    getUserName: vi.fn(),
+    getChannelName: vi.fn(),
+    getUsersInChannel: vi.fn()
+  };
+});
+
+describe('Slack message tests', () => {
```

Comment:
if im being honest, these are still like 90% unmocked tests. If anything we should rename the files to be unit and integration. and this would be integration.